recipient-resolver/src/main/java/com/redhat/cloud/notifications/recipientresolver/resolver/FetchUsersFromExternalServices.java

```diff
@@ -0,0 +1,389 @@
+package com.redhat.cloud.notifications.recipientresolver.resolver;
+
+import com.redhat.cloud.notifications.recipientresolver.config.RecipientResolverConfig;
+import com.redhat.cloud.notifications.recipientresolver.model.User;
+import com.redhat.cloud.notifications.recipientresolver.resolver.itservice.ITUserService;
+import com.redhat.cloud.notifications.recipientresolver.resolver.itservice.pojo.request.ITUserRequest;
+import com.redhat.cloud.notifications.recipientresolver.resolver.itservice.pojo.response.AccountRelationship;
+import com.redhat.cloud.notifications.recipientresolver.resolver.itservice.pojo.response.Email;
+import com.redhat.cloud.notifications.recipientresolver.resolver.itservice.pojo.response.ITUserResponse;
+import com.redhat.cloud.notifications.recipientresolver.resolver.itservice.pojo.response.Permission;
+import com.redhat.cloud.notifications.recipientresolver.resolver.mbop.MBOPService;
+import com.redhat.cloud.notifications.recipientresolver.resolver.mbop.MBOPUser;
+import com.redhat.cloud.notifications.recipientresolver.resolver.rbac.Page;
+import com.redhat.cloud.notifications.recipientresolver.resolver.rbac.RbacGroup;
+import com.redhat.cloud.notifications.recipientresolver.resolver.rbac.RbacServiceToService;
+import com.redhat.cloud.notifications.recipientresolver.resolver.rbac.RbacUser;
+import dev.failsafe.Failsafe;
+import dev.failsafe.RetryPolicy;
+import dev.failsafe.function.CheckedSupplier;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import io.netty.channel.ConnectTimeoutException;
+import io.quarkus.cache.CacheResult;
+import io.quarkus.logging.Log;
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.Response;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.resteasy.reactive.ClientWebApplicationException;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class FetchUsersFromExternalServices {
```

Comment:
The code from this class (which comes from `engine`) can be simplified a lot. Let's do that later.